nextflow-io/vscode-language-nextflow
# Bundle Mermaid.js instead of using CDN for DAG Preview

## Description

Closes https://github.com/nextflow-io/vscode-language-nextflow/issues/56

The VS Code extension currently loads `mermaid.js` from a CDN URL, which requires an internet connection to render DAG previews. This makes the feature unusable when working offline.

- Added `mermaid.js` to `package.json` dependencies
- Added `CopyWebpackPlugin` configuration to `webpack.config.js`
- Modified DAG preview webview to use local `mermaid.js` file
- Updated webview security settings to allow access to extension media files

## Note on dependencies

This PR adds two deps: mermaid itself (for obvious reasons) and `CopyWebpackPlugin`.

`CopyWebpackPlugin` (an official webpack plugin) was necessary because webpack's standard bundling process is focused on JavaScript modules and wouldn't handle the mermaid UMD file correctly. The plugin allows us to explicitly copy the pre-built mermaid.js file to our extension's output directory, preserving it in its original form which is required for the webview to load it properly. Without this plugin, the file wouldn't be available in the extension's distributed build.

## Testing

1. Build the extension locally
2. Open a Nextflow file with a DAG
3. Preview the DAG while offline - it should render correctly
